src/test/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentServiceSpec.scala

```diff
@@ -207,4 +211,182 @@ class ConsignmentServiceSpec extends AnyFlatSpec with MockitoSugar with ResetMoc
+
+    response.lastCursor should be (Some("consignment-ref2"))
+    response.consignmentEdges should have size 2
+    val edges = response.consignmentEdges
+    val cursors: List[String] = edges.map(e => e.cursor).toList
+    cursors should contain ("consignment-ref1")
+    cursors should contain ("consignment-ref2")
+
+    val consignmentRefs: List[UUID] = edges.map(e => e.node.consignmentid).toList
+    consignmentRefs should contain (consignmentId1)
+    consignmentRefs should contain (consignmentId2)
+  }
+
+  "getConsignments" should "return all the consignments up to the limit where empty cursor provided" in {
+    val consignmentId1 = UUID.fromString("20fe77a7-51b3-434c-b5f6-a14e814a2e05")
+    val consignmentId2 = UUID.fromString("fa19cd46-216f-497a-8c1d-6caaf3f421bc")
+
+    val consignmentRowParams = List(
+      (consignmentId1, "consignment-ref1", 1L),
+      (consignmentId2, "consignment-ref2", 2L)
+    )
+
+    val consignmentRows: List[ConsignmentRow] = consignmentRowParams.map(p => createConsignmentRow(p._1, p._2, p._3))
+
+    val limit = 2
+
+    val mockResponse: Future[Seq[ConsignmentRow]] = Future.successful(consignmentRows)
+    when(consignmentRepoMock.getConsignments(limit, Some(""))).thenReturn(mockResponse)
+
+    val response: PaginatedConsignments = consignmentService.getConsignments(limit, Some("")).futureValue
+
+    response.lastCursor should be (Some("consignment-ref2"))
+    response.consignmentEdges should have size 2
+    val edges = response.consignmentEdges
+    val cursors: List[String] = edges.map(e => e.cursor).toList
+    cursors should contain ("consignment-ref1")
+    cursors should contain ("consignment-ref2")
+
+    val consignmentRefs: List[UUID] = edges.map(e => e.node.consignmentid).toList
+    consignmentRefs should contain (consignmentId1)
+    consignmentRefs should contain (consignmentId2)
+  }
+
+  "getConsignments" should "return empty list and no cursor if no consignments" in {
+    val limit = 2
+    val mockResponse: Future[Seq[ConsignmentRow]] = Future.successful(Seq())
+    when(consignmentRepoMock.getConsignments(limit, Some("consignment-ref1"))).thenReturn(mockResponse)
+
+    val response: PaginatedConsignments = consignmentService.getConsignments(limit, Some("consignment-ref1")).futureValue
+
+    response.lastCursor should be (None)
+    response.consignmentEdges should have size 0
+  }
+
+  "getConsignments" should "return empty list and no cursor if limit set to '0'" in {
+    val limit = 0
+    val mockResponse: Future[Seq[ConsignmentRow]] = Future.successful(Seq())
+    when(consignmentRepoMock.getConsignments(limit, Some("consignment-ref1"))).thenReturn(mockResponse)
+
+    val response: PaginatedConsignments = consignmentService.getConsignments(limit, Some("consignment-ref1")).futureValue
+
+    response.lastCursor should be (None)
+    response.consignmentEdges should have size 0
+  }
+
+  "convertToEdges" should "return consignment edges for the given consignment rows" in {
```

Comment:
If this is a private method, then it's being implicitly tested by the other tests. You shouldn't need to test methods which aren't part of a class's public interface, partly because it makes it harder to refactor things later.